### ISSUE TITLE
Provide feedback on dynamic FFI

### ISSUE THREAD

**Issue by ofrobots** [chrisdickinson/nojs#5]
@chrisdickinson: In reference to [PLAN.md#L29](https://github.com/chrisdickinson/nojs/blame/21ba00b7ac57546fdbffd9b13d2d8f2f35ce133d/PLAN.md#L29):
> Focus on FFI. (Insert **so much :wave: handwaving :wave: here**)

I am prototyping on a [dynamic FFI](https://github.com/nodejs/api/pull/20) in V8 (hello world here: [link](https://gist.github.com/ofrobots/6cacea0863b3b286bcade6a75c7db82b)). The idea is to get the JIT compiler to generate fast JS to native bindings; similar in principle to how [LuaJIT does it](http://luajit.org/ext_ffi.html).

I would be interested in your thoughts on expectations/requirements.

**Comment by chrisdickinson** [comment]
This is really interesting — sorry it's taken me so long to get back to you. I would love to try prototyping this in nojs, if it's helpful! It should be easy to re-point [the vendored v8](https://github.com/chrisdickinson/nojs/blob/master/DEPS#L50-L51) at a forked repo that exposes a public V8 FFI API, and I'd be interested in looping that through.

**Comment by ofrobots** [comment]
Absolutely! Things are bit too raw at this point on my end, and I don't have an experimental public API yet ([see prototype here](https://codereview.chromium.org/2084663004/)). At this point I am trying to figure some example interfaces you would like ffi to support so that I could get those type semantics implemented.

Once a bit more of the type semantics are implemented, I plan to work on an experimental public API, which should make it possible to start playing with ffi here.
